apps/dashboard/lib/trpc/routers/ratelimit/query-latency-timeseries/index.ts

```diff
@@ -1,13 +1,14 @@
+import { ratelimitOverviewQueryTimeseriesPayload } from "@/app/(app)/ratelimits/[namespaceId]/_overview/components/charts/bar-chart/query-timeseries.schema";
 import { clickhouse } from "@/lib/clickhouse";
 import { db } from "@/lib/db";
 import { rateLimitedProcedure, ratelimit } from "@/lib/trpc/ratelimitProcedure";
 import { TRPCError } from "@trpc/server";
-import { z } from "zod";
+import { transformRatelimitFilters } from "./utils";
 
-export const queryDistinctIdentifiers = rateLimitedProcedure(ratelimit.update)
-  .input(z.object({ namespaceId: z.string() }))
+export const queryRatelimitLatencyTimeseries = rateLimitedProcedure(ratelimit.update)
+  .input(ratelimitOverviewQueryTimeseriesPayload)
   .query(async ({ ctx, input }) => {
-    // Get workspace
+    // Validate workspace exists and belongs to tenant
     const workspace = await db.query.workspaces
```

Comment:
would be nice to rewrite this to only load ratelimit namespace, instead of workspace + namespaces

ctx.workspace already contains a loaded workspace